src/seedu/addressbook/AddressBook.java

```diff
@@ -200,7 +200,18 @@
      */
     public static void main(String[] args) {
         showWelcomeMessage();
-        processProgramArgs(args);
+        if (args.length >= 2) {
```

Comment:
Good job. Now, your main function has different levels of abstraction, and the logic is hard to follow.